client/src/services/apiProxy.js

```diff
@@ -240,7 +241,7 @@ export function stream(url, cb, args) {
 
     function connect() {
         if (connectCb) connectCb();
-        connection = connectStream(url, cb, onFail, isJson, additionalProtocols);
+        connection = _.debounce(connectStream, 250)(url, cb, onFail, isJson, additionalProtocols);
```

Comment:
I think this won't quite do what we want. `debounce` will limit how frequently the function is called, so in this case it would rate limit the calling of `connectStream`. However, that function should only be called once when the socket is created. 

At first I was thinking the `debounce` call should move inside of `connectStream`, but after reviewing the code a little more closely, I don't think that'll work either because it could lead to missing some updates. Instead, I _think_ it's going to need to go in two places... `streamResult` and `streamResults`.

For `streamResult`, I think we'll want it to replace the line `cb(item)`. And in `streamResults` I think it would need to replace the `push` function.

Reply:
Hmmm... also, thinking about the way `_.debounce` works a little, I believe it actually returns a function that we'll need to call repeatedly. So maybe something like replacing the `push` function with something like...

```
const push = _.debounce(() => {
        const values = Object.values(results);
        cb(values);
}, 250);
```

I'm not 100% sure that's right, but I think it's something like that. 

Also, I think there are some additional parameters you can pass to debounce, but I forget exactly what they are. They may or may not be important for this use case.

Reply:
Changed according the comment above. Let me know if this works for you.